src/main/java/org/jenkinsci/plugins/ewm/nodes/ExternalWorkspaceProperty.java

```diff
@@ -4,56 +4,36 @@
 import hudson.model.Node;
 import hudson.slaves.NodeProperty;
 import hudson.slaves.NodePropertyDescriptor;
-import hudson.util.FormValidation;
 import org.jenkinsci.plugins.ewm.Messages;
-import org.kohsuke.accmod.Restricted;
-import org.kohsuke.accmod.restrictions.NoExternalUse;
 import org.kohsuke.stapler.DataBoundConstructor;
-import org.kohsuke.stapler.QueryParameter;
 
-import javax.annotation.CheckForNull;
 import javax.annotation.Nonnull;
 import java.util.List;
 
-import static hudson.Util.fixEmptyAndTrim;
 import static hudson.Util.fixNull;
-import static hudson.util.FormValidation.validateRequired;
 
 /**
- * A {@link NodeProperty} where are defined the {@link DiskNode} definitions.
+ * A {@link NodeProperty} where are defined the {@link DiskPoolNode} definitions.
  *
  * @author Alexandru Somai
  */
 public class ExternalWorkspaceProperty extends NodeProperty<Node> {
 
-    private final String diskPoolRefId;
-    private final List<DiskNode> diskNodes;
+    private final List<DiskPoolNode> diskPoolNodes;
 
     @DataBoundConstructor
-    public ExternalWorkspaceProperty(String diskPoolRefId, List<DiskNode> diskNodes) {
-        this.diskPoolRefId = fixEmptyAndTrim(diskPoolRefId);
-        this.diskNodes = fixNull(diskNodes);
-    }
-
-    @CheckForNull
-    public String getDiskPoolRefId() {
-        return diskPoolRefId;
+    public ExternalWorkspaceProperty(List<DiskPoolNode> diskPoolNodes) {
+        this.diskPoolNodes = fixNull(diskPoolNodes);
     }
 
     @Nonnull
-    public List<DiskNode> getDiskNodes() {
-        return diskNodes;
+    public List<DiskPoolNode> getDiskPoolNodes() {
+        return diskPoolNodes;
```

Comment:
It exposes the internal implementation (I would expect FindBugs to grumble about it). Wrapping by Collectioms.unmodifyableList() may be a good approach